### PR TITLE
Add complex power (.powc()) for T: float.

### PR DESCRIPTION
**Pull request by ticki**

### CODE REVIEW

**Comment by Ticki** [comment]
Highfive, my fellow bot.

**Comment by alexcrichton** [comment]
Can you also add some tests for this?

**Comment by Ticki** [comment]
I'll do.

**Comment by hauleth** [comment]
@Ticki what about tests?

File: src/complex.rs [hunk 1]
```diff
@@ -100,6 +100,18 @@ impl<T: Clone + Float + Num> Complex<T> {
     pub fn from_polar(r: &T, theta: &T) -> Complex<T> {
         Complex::new(*r * theta.cos(), *r * theta.sin())
     }
+
+    /// Raises self to a complex power `exp`
+    #[inline]
+    pub fn powc(&self, exp: &Complex<T>) -> Complex<T> {
+      // Formula: (r e^(ti))^(a+bi) = (r^a e^(-bt)) e^((ln(r)b+at)i)
+      let (r, t) = exp.to_polar();
```
Comment (cuviper) [inline src/complex.rs]: `self` is the base that should be made polar, not `exp`.

File: src/complex.rs [hunk 2]
```diff
@@ -100,6 +100,18 @@ impl<T: Clone + Float + Num> Complex<T> {
...
+
+      let r2 = r.powf(exp.re) * (-exp.im * t).exp();
+      let t2 = r.ln() * self.im + self.re * t;
```
Comment (cuviper) [inline src/complex.rs]: `t2` needs to be computed from `exp.im` and `exp.re`, not from `self`.

**Comment by cuviper** [comment]
Superseded by #137.

**Pull request closed without merging**